File: 02_Object_Oriented_Programming/04_This_Keyword/02_Constructor_Chaining/python/person.py
class Person:
    # Python doesn't support multiple constructors directly,
    # so we simulate constructor chaining with default parameters
    def __init__(self, name, age=0):
        self.name = name
        self.age = age

    def display(self):
        print(f"Name: {self.name}, Age: {self.age}")

File: 02_Object_Oriented_Programming/04_This_Keyword/02_Constructor_Chaining/javascript/person.js
class Person {
    // JavaScript doesn't support multiple constructors directly,
    // so we simulate constructor chaining with default parameters
    constructor(name, age = 0) {
        this.name = name;
        this.age = age;
    }

    display() {
        console.log("Name: " + this.name + ", Age: " + this.age);
    }
}

module.exports = Person;

